internal/staging/stdaddress/address.go

```diff
@@ -0,0 +1,434 @@
+// public key in the compressed format or the associated coins will NOT be
+// redeemable.
+//
+// NOTE: Version 0 scripts are the only currently supported version.
+func NewAddressPubKeyHashSchnorrSecp256k1(scriptVersion uint16, pkHash []byte,
+	params AddressParams) (Address, error) {
+
+	switch scriptVersion {
+	case 0:
+		return NewAddressPubKeyHashSchnorrSecp256k1V0(pkHash, params)
+	}
+
+	str := fmt.Sprintf("pubkey hash addresses for version %d are not "+
+		"supported", scriptVersion)
+	return nil, makeError(ErrUnsupportedScriptVersion, str)
+}
+
+// NewAddressScriptHashFromHash returns an address that represents a payment
+// destination which imposes an encumbrance that requires a script that hashes
+// to the provided script hash along with all of the encumbrances that script
+// itself imposes.  The script is commonly referred to as a redeem script.
+//
+// For version 0 scripts, the provided script hash must be 20 bytes and is
+// expected to be the Hash160 of the associated redeem script.
+//
+// See NewAddressScriptHash for a variant that accepts the redeem script instead
+// of its hash.  It can be used as a convenience for callers that have the
+// redeem script available.
+//
+// NOTE: Version 0 scripts are the only currently supported version.
+func NewAddressScriptHashFromHash(scriptVersion uint16, scriptHash []byte,
+	params AddressParams) (Address, error) {
+
+	switch scriptVersion {
+	case 0:
+		return NewAddressScriptHashV0FromHash(scriptHash, params)
+	}
+
+	str := fmt.Sprintf("script hash addresses for version %d are not "+
+		"supported", scriptVersion)
+	return nil, makeError(ErrUnsupportedScriptVersion, str)
+}
+
+// NewAddressScriptHash returns an address that represents a payment destination
+// which imposes an encumbrance that requires a script that hashes to the same
+// value as the provided script along with all of the encumbrances that script
+// itself imposes.  The script is commonly referred to as a redeem script.
+//
+// See NewAddressScriptHashFromHash for a variant that accepts the hash of the
+// script directly instead of the script.  It can be useful to callers that
+// either already have the script hash available or do not know the associated
+// script.
+//
+// NOTE: Version 0 scripts are the only currently supported version.
+func NewAddressScriptHash(scriptVersion uint16, redeemScript []byte,
+	params AddressParams) (Address, error) {
+
+	switch scriptVersion {
+	case 0:
+		return NewAddressScriptHashV0(redeemScript, params)
+	}
+
+	str := fmt.Sprintf("script hash addresses for version %d are not "+
+		"supported", scriptVersion)
+	return nil, makeError(ErrUnsupportedScriptVersion, str)
+}
+
+// probablyBase58Addr returns true when the provided string only contains runes
+// in the base58 alphabet used by Decred for version 0 addresses.
+func probablyBase58Addr(s string) bool {
+	// The modified base58 alphabet used by Decred for version 0 addresses is:
```

Comment:
This should check the length up front since the set of supported v0 addresses always produce addresses within a certain length range.  That will help potentially misidentifying other formats which are a subset.

Reply:
Done.